File: Random/Arrays & Hashing/largest-number-at-least-twice-of-others.js


var maximumValue = function(nums){
    let maxValue = 0;
    for(let index in nums){
        if(nums[index]>maxValue){
            maxValue = nums[index]
        }
    }
    return maxValue
}

var dominantIndex = function(nums) {
    let maxItem = maximumValue(nums);
    let idx = nums.indexOf(maxItem);
    let count = 0;
    nums.splice(idx,1)
    for(let val of nums){
        if(maxItem>= val*2){
            count +=1
        }
    }
    if(count==nums.length){
        return idx
    }else{
        return -1
    }
    
    
};
dominantIndex([3,6,1,0])

File: Random/Arrays & Hashing/largest-number-at-least-twice-of-others.py

# time comeplexy = o(n)
class Solution(object):

        
    def dominantIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        maximum_value = max(nums)
        idx = nums.index(maximum_value)
        nums.remove(maximum_value)
        count = 0
        for val in nums:
            if maximum_value>=val*2:
                count +=1 
        if count == len(nums):
            return idx
        return -1       

s = Solution()
print(s.dominantIndex([3,6,1,0]))